src/main/java/it/gov/pagopa/pu/sil/service/legacyauth/SilLegacyJwtAuthService.java

```diff
@@ -1,13 +1,48 @@
 package it.gov.pagopa.pu.sil.service.legacyauth;
 
+import com.auth0.jwt.HeaderParams;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTCreator;
+import com.auth0.jwt.algorithms.Algorithm;
 import it.gov.pagopa.pu.auth.dto.generated.AccessToken;
 import it.gov.pagopa.pu.organization.dto.generated.SilServiceLegacyJwtAuthConfig;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
 @Service
 public class SilLegacyJwtAuthService {
+  public static final String ACCESS_TOKEN_TYPE = "at+JWT";
+
+  private final Integer expirationTimeInSeconds;
+
+  public SilLegacyJwtAuthService(
+    @Value("${legacy-auth.default-expiration-seconds}") Integer expirationTimeInSeconds) {
```

Comment:
The `default-expiration-seconds` has been configured with a low value in order to configure a not too long caching period for legacyBasic token, where the expiration is not handled by us...
Here, I would set a new property `jwt-legacy-expiration-seconds`, which could be configured with a longer value (eg 1 hour).

ps: we could think to rename `default-expiration-seconds` into `basic-legacy-expiration-seconds`